### PR TITLE
[fix] tril & tril : out of bound check

### LINKED ISSUE #83326: torch.tril lacks of checking 'out of bound' on cuda
### 🐛 Describe the bug

```
results = dict()
import torch
arg_1 = torch.rand([4], dtype=torch.float32)
try:
  results["res_cpu"] = torch.tril(arg_1,)
except Exception as e:
  results["err_cpu"] = "ERROR:"+str(e)
arg_2 = arg_1.clone().cuda()
try:
  results["res_gpu"] = torch.tril(arg_2,)
except Exception as e:
  results["err_gpu"] = "ERROR:"+str(e)
print(results)
```

The result is {'err_cpu': 'ERROR:Dimension out of range (expected to be in range of [-1, 0], but got -2)', 'res_gpu': tensor([0.6544, 0.0000, 0.0000, 0.0000], device='cuda:0')}.

### Versions

pytorch: 1.8.1
python: 3.8.3
os: win11

cc @ngimel

### PR DESCRIPTION
**Pull request by khushi-411**
Fixes #83326

### CODE REVIEW

**Comment by ngimel** [comment]
These checks are better added to meta function https://github.com/pytorch/pytorch/blob/c6ae8d300dbe03901cf77f738c5dcd915fc4ca6b/aten/src/ATen/native/TriangularOps.cpp#L26, then they don't need to be done independently for cpu and cuda.

**Comment by khushi-411** [comment]
Indeed a better way. Thanks, @ngimel! Have updated it.

File: test/functorch/test_vmap.py [hunk 1]
```diff
@@ -3341,6 +3343,8 @@ def test_vmap_exhaustive(self, device, dtype, op):
         xfail('tensor_split'),
         xfail('to_sparse'),
         xfail('vdot'),
+        xfail('tril'),
```
Comment (ngimel) [inline test/functorch/test_vmap.py:3346]: what happens here?

**Review by ngimel — APPROVED** [review]

File: test/functorch/test_vmap.py (hunk 1 above)
Comment (khushi-411) [reply test/functorch/test_vmap.py]: added a comment.

This raises `AssertionError: Exception not raised`. I think most likely missing a check in their def of their batching rule.

**Pull request closed without merging**